homeassistant/config.py

```diff
@@ -743,13 +743,18 @@ def async_process_component_config(
             async_log_exception(ex, domain, config, hass)
             return None
 
-    elif hasattr(component, 'PLATFORM_SCHEMA'):
+    elif (hasattr(component, 'COMPONENT_SCHEMA') or
```

Comment:
It's weird that the word `COMPONENT_SCHEMA` triggers platform validation.

The idea was: Oh it has a platform schema, let's treat the config as being a list of platforms that all have to validate. Can't we call it `PLATFORM_BASE_SCHEMA`, given that it is the schema that is the basis for all platform schemas of that entity component ?

Reply:
no no, we flip it. The BASE is not what normal platform extends from, BASE is validating that the bare minimum of a valid platform config is validated.

Reply:
OK, that will be easier!

Just to double confirm, this is what you mean:
```python
PLATFORM_SCHEMA = vol.Schema({
    vol.Required(CONF_PLATFORM): string,
    vol.Optional(CONF_SCAN_INTERVAL): time_period
})

PLATFORM_SCHEMA_BASE = PLATFORM_SCHEMA.extend({
}, extra=vol.ALLOW_EXTRA)
```